client/app/visualizations/vega/Renderer.jsx

```diff
@@ -0,0 +1,249 @@
+import stringify from 'json-stringify-pretty-compact';
+import ResizeObserver from 'resize-observer-polyfill';
+import { isObject } from 'lodash';
+import React from 'react';
+import Vega from 'react-vega';
+import * as vl from 'vega-lite';
+// import * as YAML from 'js-yaml';
+import { Handler } from 'vega-tooltip';
+import { Alert, Icon } from 'antd';
+import LZString from 'lz-string';
+import memoize from 'memoize-one';
```

Comment:
[`_.memoize`](https://lodash.com/docs/4.17.11#memoize)? [`useMemo`](https://reactjs.org/docs/hooks-reference.html#usememo)?

Reply:
`_.memoize` does not cache only one return value, having higher risk of memory leaks.
`useMemo` does not work with class components. I don't see the added benefits of rewriting my class to a function component.

`memoize-one` only has 45 LOC.